deepwell/src/services/email/service.rs

```diff
@@ -0,0 +1,95 @@
+/*
+ * services/email/service.rs
+ *
+ * DEEPWELL - Wikijump API provider and database manager
+ * Copyright (C) 2019-2023 Wikijump Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use super::prelude::*;
+
+#[derive(Debug)]
+pub struct EmailService;
+
+impl EmailService {
+    /// Validates an email through the MailCheck API.
+    pub async fn validate(email: &String) -> Result<EmailValidationOutput> {
+        // Sends a GET request to the MailCheck API and deserializes the response.
+        let mailcheck = surf::get(format!("https://api.mailcheck.ai/email/{email}"))
+            .send()
+            .await?
+            .body_json::<MailCheckResponse>()
+            .await?;
+
+        // Create the output with default parameters.
+        let mut output = EmailValidationOutput::default();
+
+        // Check request status.
+        match mailcheck.status {
+            // Valid request.
+            200 => {}
+
+            // Invalid request.
+            400 => {
+                tide::log::error!(
+                    "MailCheck API returned status 400: {:?}",
```

Comment:
Since we're checking individual status codes rather than just doing `MailCheck API returned {}`, I'd rather these messages be like `MailCheck API request fails with bad response:` and `MailCheck API ratelimited:`